android/app/src/main/java/com/graphhopper/maps/MainActivity.java

```diff
@@ -78,11 +76,13 @@ public void onRequestPermissionsResult(int requestCode, String[] permissions, in
 
     private void requestLocationUpdates() {
         // Request permission, if it is not yet provided
-        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED
+                && ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
             // Requires at least Marshmallow
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-                requestPermissions(new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, REQUEST_LOCATION);
-            }
+//            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                // TODO NOW infinite loop if no GPS was enabled -> why is this necessary at all?
+//                requestPermissions(new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, REQUEST_LOCATION);
+//            }
```

Comment:
I can have a look at what is happening here. 

It is required because location permissions are runtime permissions starting with Android 7 or Android 8, you can't just add the permission to the Manifest. You need to explicitly ask the user.

This repo is not using the latest capacitor, with capacitor V4 the permission handling has been completely updated, so updating to capacitor V4 might be a good idea. This would probably make the permission handling easier.

Reply:
Oh, I just saw that you updated this to the latest capacitor in this PR 😆 - well done.

With Capacitor 4 you would write/use a plugin that requests the permissions. There is a ready made plugin available: https://capacitorjs.com/docs/apis/geolocation#requestpermissions

So I think we should use this plugin here.

Reply:
One way to do this would be to extend the app.js [here](https://github.com/boldtrn/graphhopper-maps-capacitor/blob/main/src/app.js#L43). Here we would wrap the watch position method of the geolocation plugin. Then we call this method from gh-maps, similar to what we do for TTS already.

Reply:
Yes, I think we could remove the Java code, if we would be using the plugin.